docs/unions.md

```diff
@@ -0,0 +1,119 @@
+:::{php:namespace} Atk4\Data
+:::
+
+(Unions)=
+
+# Model Unions
+
+:::{php:class} Model\UnionModel
+:::
+
+In some cases data from multiple models need to be combined. In this case the UnionModel model comes very handy.
+In the case used below Client model schema may have multiple invoices and multiple payments. Payment is not related to the invoice.:
+
+```
+class Client extends \Atk4\Data\Model {
+    public $table = 'client';
+
+    protected function init(): void
+    {
+        parent::init();
+
+        $this->addField('name');
+
+        $this->hasMany('Payment');
+        $this->hasMany('Invoice');
+    }
+}
+```
+
+(see tests/ModelUnionTest.php, tests/Model/Client.php, tests/Model/Payment.php and tests/Model/Invoice.php files).
+
+## Union Model Definition
+
+Normally a model is associated with a single table. Union model can have multiple nested models defined and it fetches
+results from that. As a result, Union model will have no "id" field. Below is an example of inline definition of Union model.
+The Union model can be separated in a designated class and nested model added within the init() method body of the new class:
+
+```
+$unionPaymentInvoice = new \Atk4\Data\Model\Union();
```

Comment:
should be `\Atk4\Data\Model\UnionModel()`

Reply:
👍